### ISSUE TITLE
Define navigation button as a heading in code

### ISSUE THREAD

**Issue by Edavda** [Altinn/app-frontend-react#3283]
Saja from SSB conducted a quick user test of the navigation structure with users that rely on screen readers. Based on the results from the test, they concluded that the navigation button should be set as a heading `(<h2> <button>...</button></h2>)` and placed after the main content of the page. 

Here is the summary of the test results and conclusion.

> Resultatet viste en enighet i hovedsakelig to ting:
Skjemanavigasjon var forventet å ligge etter hovedinnholdet
Ønskemål at det skulle være en overskrift, slik at det er enklere å finne (enn kun beskrivelse av en knapp)
 
For begge deltakerne tok det tid til å finne skjemaoversikten når den lå skjult bak en knapp beskrevet som «Skjemasider». Begge deltakerne startet å navigere seg på siden ved å hoppe direkte til hovedinnholdet, og derfra gå igjennom siden ved å hoppe mellom overskriftene. Ingen av deltakerne navigerte seg før hovedinnholdet, inntil jeg etterhvert instruerte dem til å gjøre det.
Det var også problemer med de utvidbare knappene i skjemaoversikten da deltakerne ikke fikk tilgang til informasjonen under i det hele tatt.
På bakgrunn av dette, ville jeg da anbefalt å legge selve innholdet plassert etter hovedinnholdet i koden (kan fortsatt ligge plassert øverst visuelt), og definere tittelen på knappen som en overskrift, slik som nevnt tidligere. (<h2> <button>...</button></h2>).

**Comment by olemartinorg** [comment]
@Edavda Vet du hvilket skjema dette gjelder, eller hvordan dette skjemaet ble satt opp? Med andre ord, gjelder dette ny navigasjon (som vises på siden av hovedinnholdet), eller spesifikt `NavigationButtons`-komponenten?

Dette indikerer for meg at det er snakk om ny navigasjon som vises på siden:
> Ingen av deltakerne navigerte seg før hovedinnholdet, inntil jeg etterhvert instruerte dem til å gjøre det.

Mens dette beskriver jo hvordan man vanligvis setter opp `NavigationButtons`-komponenten i dag, dog den er ikke pakket i h2.
> they concluded that the navigation button should be set as a heading (`<h2> <button>...</button></h2>`) and placed after the main content of the page.

Jeg tror heller ikke at h2 nødvendigvis er det helt riktige å bruke for å indikere navigasjon, for det har vi jo heller `<nav>` eller [navigasjonsrollen](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/navigation_role). Jeg mener vi har den for den nye navigasjonen, men neppe for de gamle navigasjonsknappene. En skjermleser burde nesten ha støtte for å hoppe rett til navigasjonselementer, men jeg finner ingen umiddelbar oppskrift på hvordan man gjør det i f.eks. NVDA. 🤔

**Comment by Edavda** [comment]
@olemartinorg 
Dette var en brukertest som SSB hadde satt opp med ny navigasjon. Jeg ser at det ikke kom tydlig fram i beskrivelsen her 🙈

Vi hadde en diskusjon med de om hvordan den nye navigasjonskomponenten burde struktureres for at den skal enklere å bruke for skjermlesere, men vi klarte ikke å konkludere under møtet, så Saja gjennomførte en test og kom med en anbefaling.

Jeg husker vi diskuterte å legge det i `nav` eller `aside` mtp på plasseringen av navigasjonsmenyen og om dette skulle komme før eller etter hovedinnholdet. Her har ikke jeg noen sterke meninger om hva som blir mest riktig å bruke, men det viktigste er at vi får ivaretatt en best mulig flyt for skjermlesere som bruker navigasjonen.
